hack/make-rules/test.sh

```diff
@@ -32,28 +32,18 @@ export KUBE_CACHE_MUTATION_DETECTOR
 KUBE_PANIC_WATCH_DECODE_ERROR="${KUBE_PANIC_WATCH_DECODE_ERROR:-true}"
 export KUBE_PANIC_WATCH_DECODE_ERROR
 
-kube::test::find_dirs() {
+kube::test::find_packages() {
   (
     cd "${KUBE_ROOT}"
-    find -L . -not \( \
-        \( \
-          -path './_artifacts/*' \
-          -o -path './_output/*' \
-          -o -path './cmd/kubeadm/test/*' \
-          -o -path './contrib/podex/*' \
-          -o -path './release/*' \
-          -o -path './target/*' \
-          -o -path './test/e2e/e2e_test.go' \
-          -o -path './test/e2e_node/*' \
-          -o -path './test/e2e_kubeadm/*' \
-          -o -path './test/integration/*' \
-          -o -path './third_party/*' \
-          -o -path './staging/*' \
-          -o -path './vendor/*' \
-        \) -prune \
-      \) -name '*_test.go' -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
-
-    find ./staging -name '*_test.go' -not -path '*/test/integration/*' -prune -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
+    # shellcheck disable=SC2046
+    go list -find $(go work edit -json | jq -r '.Use[].DiskPath + "/..."') \
+        | grep -vE \
+            -e '^k8s.io/kubernetes/third_party(/.*)?$' \
+            -e '^k8s.io/kubernetes/cmd/kubeadm/test(/.*)?$' \
+            -e '^k8s.io/kubernetes/test/e2e$' \
+            -e '^k8s.io/kubernetes/test/e2e_node(/.*)?$' \
+            -e '^k8s.io/kubernetes/test/e2e_kubeadm(/.*)?$' \
+            -e '^k8s.io/.*/test/integration(/.*)?$'
```

Comment:
```suggestion
    local -a all_module_package_patterns
    kube::util::read-array all_module_package_patterns < <(go work edit -json | jq -r '.Use[].DiskPath + "/..."')
    go list -find "${all_module_package_patterns[@]}" \
        | grep -vE \
            -e '^k8s.io/kubernetes/third_party(/.*)?$' \
            -e '^k8s.io/kubernetes/cmd/kubeadm/test(/.*)?$' \
            -e '^k8s.io/kubernetes/test/e2e$' \
            -e '^k8s.io/kubernetes/test/e2e_node(/.*)?$' \
            -e '^k8s.io/kubernetes/test/e2e_kubeadm(/.*)?$' \
            -e '^k8s.io/.*/test/integration(/.*)?$'
```

naming is hard, there's probably a clearer yet terser variable name for all_module_package_patterns, but I think overall this is a little tiny bit more obvious without digging into `go work edit` and should pass shellcheck?

Reply:
The main point being to a shellcheck?

Reply:
and to be a little bit more obvious to a reader that isn't deeply familiar with `go work edit`